OrbitGl/TimeGraphLayout.cpp

```diff
@@ -85,3 +85,17 @@ bool TimeGraphLayout::DrawProperties() {
 
   return needs_redraw;
 }
+
+void TimeGraphLayout::VerticalZoom(double scale) {
+  m_TextBoxHeight *= scale;
+  m_CoresHeight *= scale;
+  m_EventTrackHeight *= scale;
+  m_GraphTrackHeight *= scale;
+  m_SpaceBetweenCores *= scale;
+  m_SpaceBetweenTracks *= scale;
+  m_SpaceBetweenTracksAndThread *= scale;
+  m_TrackTabHeight *= scale;
+  m_RoundingRadius *= scale;
+  m_TrackBottomMargin *= scale;
+  m_TrackTopMargin *= scale;
+}
```

Comment:
Scale should be kept as it's own member variable and all the other variables should be kept independent of the scale. The getters should return the result of the scale multiplied by the value. The problem is that the floating point values will drift over time and we won't be able to return to the exact initial values as floating point error will accumulate forever. Also, having scale separate will allow us to have an interface to set scale directly without having to go through many increments. A reset could then be implemented easily by calling SetScale(1.f);

Reply:
You are completely right